Document migration for sendclientreports (#4971)
getsentry/sentry-docs: src/platforms/dart/migration.mdx

description: "Migrate between versions of Sentry's SDK for Dart."
---

## Migrating From `sentry` `6.5.x` to `sentry` `6.6.x`

> - `SentryOptions#sendClientReports` is now enabled by default, to disable it, see the code snippet bellow.

Reviewer: ```suggestion
- `SentryOptions#sendClientReports` is now enabled by default. To disable it, use the code snippet below:
```

```dart
import 'package:sentry/sentry.dart';

Future<void> main() async {
  await Sentry.init((options) => options.sendClientReports = false;
}
```

### Sentry Self-hosted Compatibility

> - Starting with version `6.6.0` of `sentry`, [Sentry's version >= v21.9.0](https://github.com/getsentry/self-hosted/releases) is required. This only applies to self-hosted Sentry. If you are using [sentry.io](https://sentry.io), no action is needed.

Reviewer 2: ```suggestion
- Starting with version `6.6.0` of `sentry`, [Sentry's version >= v21.9.0](https://github.com/getsentry/self-hosted/releases) is required or you have to manually disable sending client reports via the `sendClientReports` option. This only applies to self-hosted Sentry. If you are using [sentry.io](https://sentry.io), no action is needed.
```

Reviewer 2: Do we also mention that you can get around the self hosted upgrade by disabling client reports here?

> - Starting with version `6.6.0` of the `sentry`, [Sentry's version >= v21.9.0](https://github.com/getsentry/self-hosted/releases) is required. This only applies to self-hosted Sentry. If you are using [sentry.io](https://sentry.io), no action is needed.

Reviewer: ```suggestion
- Starting with version `6.6.0` of `sentry`, [Sentry's version >= v21.9.0](https://github.com/getsentry/self-hosted/releases) is required. This only applies to self-hosted Sentry. If you are using [sentry.io](https://sentry.io), no action is needed.
```
